numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -4795,147 +4790,191 @@ set_flaginfo(PyObject *d)
-    if (npy_ma_str_axis1 == NULL) {
+    npy_ma_str.axis1 = PyUnicode_InternFromString("axis1");
+    if (npy_ma_str.axis1 == NULL) {
         return -1;
     }
-    npy_ma_str_axis2 = PyUnicode_InternFromString("axis2");
-    if (npy_ma_str_axis2 == NULL) {
+    npy_ma_str.axis2 = PyUnicode_InternFromString("axis2");
+    if (npy_ma_str.axis2 == NULL) {
         return -1;
     }
-    npy_ma_str_like = PyUnicode_InternFromString("like");
-    if (npy_ma_str_like == NULL) {
+    npy_ma_str.like = PyUnicode_InternFromString("like");
+    if (npy_ma_str.like == NULL) {
         return -1;
     }
-    npy_ma_str_numpy = PyUnicode_InternFromString("numpy");
-    if (npy_ma_str_numpy == NULL) {
+    npy_ma_str.numpy = PyUnicode_InternFromString("numpy");
+    if (npy_ma_str.numpy == NULL) {
         return -1;
     }
-    npy_ma_str_where = PyUnicode_InternFromString("where");
-    if (npy_ma_str_where == NULL) {
+    npy_ma_str.where = PyUnicode_InternFromString("where");
+    if (npy_ma_str.where == NULL) {
         return -1;
     }
     /* scalar policies */
-    npy_ma_str_convert = PyUnicode_InternFromString("convert");
-    if (npy_ma_str_convert == NULL) {
+    npy_ma_str.convert = PyUnicode_InternFromString("convert");
+    if (npy_ma_str.convert == NULL) {
         return -1;
     }
-    npy_ma_str_preserve = PyUnicode_InternFromString("preserve");
-    if (npy_ma_str_preserve == NULL) {
+    npy_ma_str.preserve = PyUnicode_InternFromString("preserve");
+    if (npy_ma_str.preserve == NULL) {
         return -1;
     }
-    npy_ma_str_convert_if_no_array = PyUnicode_InternFromString("convert_if_no_array");
-    if (npy_ma_str_convert_if_no_array == NULL) {
+    npy_ma_str.convert_if_no_array = PyUnicode_InternFromString("convert_if_no_array");
+    if (npy_ma_str.convert_if_no_array == NULL) {
         return -1;
     }
-    npy_ma_str_cpu = PyUnicode_InternFromString("cpu");
-    if (npy_ma_str_cpu == NULL) {
+    npy_ma_str.cpu = PyUnicode_InternFromString("cpu");
+    if (npy_ma_str.cpu == NULL) {
         return -1;
     }
-    npy_ma_str_dtype = PyUnicode_InternFromString("dtype");
-    if (npy_ma_str_dtype == NULL) {
+    npy_ma_str.dtype = PyUnicode_InternFromString("dtype");
+    if (npy_ma_str.dtype == NULL) {
         return -1;
     }
-    npy_ma_str_array_err_msg_substr = PyUnicode_InternFromString(
+    npy_ma_str.array_err_msg_substr = PyUnicode_InternFromString(
             "__array__() got an unexpected keyword argument 'copy'");
-    if (npy_ma_str_array_err_msg_substr == NULL) {
+    if (npy_ma_str.array_err_msg_substr == NULL) {
+        return -1;
+    }
+    npy_ma_str.out = PyUnicode_InternFromString("out");
+    if (npy_ma_str.out == NULL) {
         return -1;
     }
-    npy_ma_str___dlpack__ = PyUnicode_InternFromString("__dlpack__");
-    if (npy_ma_str___dlpack__ == NULL) {
+    npy_ma_str.__dlpack__ = PyUnicode_InternFromString("__dlpack__");
+    if (npy_ma_str.__dlpack__ == NULL) {
```

Comment:
> I also think we should move this into it's own file.

Sorry, just to clarify - what should be moved into its own file? Filling all the static structs? Or just the string interning?

For the former, fair enough, that occured to me, and it would separate multiarraymodule.h from the global statics.

Reply:
I would say all of the static ones in one file maybe?  Seemed worthwhile to try to use this opportunity to clean things up a bit nicer (but I am happy to help with it!)